File: backend/spring-boot/src/main/java/org/bootstrapbugz/api/shared/generic/crud/CrudService.java
package org.bootstrapbugz.api.shared.generic.crud;

import org.bootstrapbugz.api.shared.payload.dto.PageableDTO;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
public interface CrudService<T, U> {
  T create(U saveRequest);

  PageableDTO<T> findAll(Pageable pageable);

  T findById(Long id);

  T update(Long id, U saveRequest);

  void delete(Long id);
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/shared/generic/crud/CrudService.kt
package org.bootstrapbugz.api.shared.generic.crud

import org.springframework.data.domain.Pageable

interface CrudService<T, U> {
  fun create(request: U): T

  fun findAll(pageable: Pageable): List<T>

  fun findById(id: Long): T

  fun update(id: Long, request: U): T

  fun delete(id: Long)
}
